### PR TITLE
feat: flag delete formula

### LINKED ISSUE #716: Support flags for rit delete formula
**What would you like to be added**:
Support flags for `rit delete formula`

**Why is this needed**:
Improve integrability and standardize flags over stdin

### PR DESCRIPTION
**Pull request by lucasdittrichzup**
### Description
Adding flag support to `rit delete formula` command. Some of the refactoring done as well in this PR:

- Added tests to support flags and changed tests to work with a tmp directory instead of only mocks.

Close #716 

### How to verify it
- Run `rit delete formula --workspace=<workspace name> --formula=<rit test formula>` and check if has been deleted succesfully.
- Run `rit delete formula` and check if the delete confirmation works.

### Changelog
Update delete formula for input flags

### CODE REVIEW

File: pkg/cmd/delete_formula.go [hunk 1]
```diff
@@ -41,6 +43,8 @@ const (
 	srcDir                     = "src"
 	questionSelectFormulaGroup = "Select a formula or group: "
 	optionOtherFormula         = "Another formula"
+	workspaceFlagName          = "workspace"
+	formulaFlagName            = "formula"
```
Comment (GuillaumeFalourd) [inline pkg/cmd/delete_formula.go:47]: By formula Name, do you mean formula command?

Comment (lucasdittrichzup) [reply pkg/cmd/delete_formula.go:47]: yes

File: pkg/cmd/delete_formula.go [hunk 2]
```diff
@@ -311,7 +411,7 @@ func (d deleteFormulaCmd) recreateTreeJSON(workspace string) error {
 	}
 
 	jsonString, _ := json.MarshalIndent(localTree, "", "\t")
-	pathLocalTreeJSON := filepath.Join(d.ritchieHomeDir, "repos", "local", "tree.json")
+	pathLocalTreeJSON := filepath.Join(workspace, "tree.json")
```
[the remark this answers is not in the document]
Comment (lucasdittrichzup) [reply pkg/cmd/delete_formula.go:414]: I agree and I'll fix it

**Review by GuillaumeFalourd — APPROVED** [review]
Good work, well done! 👍🏼

**Pull request merged**